### PR TITLE
feat: redirect on 401 auth

### PR DESCRIPTION
**Pull request by fmiras**
# What? 
...

# Why? 
...

### CODE REVIEW

File: packages/shared/index.ts [hunk 1]
```diff
@@ -72,32 +72,34 @@ export async function initShared(container: HTMLElement): Promise<ETHEREUM_NETWO
...
       const payload: any = await auth.getAccessTokenData()
       user_id = payload.user_id
     } catch (e) {
-      console['error'](e)
+      defaultLogger.error(e)
+      if (knownTLDs.includes(getTLD())) {
+        window.location.href = `https://explorer.decentraland.${getTLD()}`
+      }
       console['groupEnd']()
       throw new Error('Authentication error. Please reload the page to try again. (' + e.toString() + ')')
```
Comment (matiasbargas) [inline packages/shared/index.ts:85]: I think we can delete this exception and log it with the previous `defaultLogger.error(e)` instead

Comment (fmiras) [reply packages/shared/index.ts:85]: This error is going to still exists if the application is not hosted at `explorer.decentraland.x` e.g. localhost, new entrypoints or any other thing. If we just log an infinite spinner will take place

Comment (eordano) [reply packages/shared/index.ts:85]: Doesn't this swipe out any `?position=1,1` query parameters?

**Review by eordano — CHANGES_REQUESTED** [review]

File: packages/shared/index.ts [hunk 2]
```diff
@@ -72,32 +72,34 @@ export async function initShared(container: HTMLElement): Promise<ETHEREUM_NETWO
...
     await auth.login(container)
-
     try {
       const payload: any = await auth.getAccessTokenData()
       user_id = payload.user_id
     } catch (e) {
-      console['error'](e)
+      defaultLogger.error(e)
+      if (knownTLDs.includes(getTLD())) {
+        window.location.href = `https://explorer.decentraland.${getTLD()}`
```
Comment (menduz) [inline packages/shared/index.ts]: This change should not be handled from here. This is a initShared method, should not handle redirections. Please move the logic to the caller entry point

**Review by menduz — CHANGES_REQUESTED** [review]

File: packages/unity-interface/initializer.ts [hunk 3]
```diff
@@ -45,7 +45,13 @@ const engineInitialized = future()
 export async function initializeUnity(container: HTMLElement): Promise<InitializeUnityResult> {
   _container = container
 
-  await initShared(container)
+  try {
+    await initShared(container)
+  } catch (error) {
+    if (error.message.includes('Authentication error') === '' && knownTLDs.includes(getTLD())) {
+      window.location.href = `https://explorer.decentraland.${getTLD()}${window.location.search}`
```
Comment (menduz) [inline packages/unity-interface/initializer.ts]: Move this one level upper please, to the unity.ts file in the entry points. Here it breaks CLI and Editor

Also `window.location.reload()` makes the trick

**Review by menduz — CHANGES_REQUESTED** [review]

File: packages/unity-interface/initializer.ts (hunk 3 above)
Comment (menduz) [reply packages/unity-interface/initializer.ts]: https://developer.mozilla.org/en-US/docs/Web/API/Location/reload

**Review by menduz — APPROVED** [review]

**Pull request merged**